src/main/java/org/sagebionetworks/bridge/spring/controllers/AdherenceController.java

```diff
@@ -91,4 +93,38 @@ public StatusMessage updateAdherenceRecords(@PathVariable String studyId, @PathV
         
         return service.getAdherenceRecords(session.getAppId(), search);
     }
+
+    @DeleteMapping("/v5/studies/{studyId}/participants/self/adherence")
+    public StatusMessage deleteAdherenceRecordsForSelf(@PathVariable String studyId) {
+        UserSession session = getAuthenticatedAndConsentedSession();
+
+        AdherenceRecordList recordsList = parseJson(AdherenceRecordList.class);
+        for (AdherenceRecord oneRecord : recordsList.getRecords()) {
+            oneRecord.setAppId(session.getAppId());
+            oneRecord.setUserId(session.getId());
+            oneRecord.setStudyId(studyId);
+        }
+//        TODO: Build deletion method in service
+//        service.deleteAdherenceRecords(session.getAppId(), recordsList);
+        return DELETED_MSG;
+    }
+
+    @DeleteMapping("/v5/studies/{studyId}/participants/{userIdToken}/adherence")
+    public StatusMessage deleteAdherenceRecords(@PathVariable String studyId, @PathVariable String userIdToken) {
+//        TODO: Verify roles with access to Delete. Currently same as Update.
+        UserSession session = getAuthenticatedSession(RESEARCHER, STUDY_COORDINATOR);
```

Comment:
Yep, seems good. Some day we're going to have a way for testers to do this kind of thing too.